queries.sql

```diff
@@ -12,4 +12,41 @@ SELECT * FROM animals WHERE weight >= 10.4 AND weight <= 17.3;
 
 SELECT name,escape_attempts FROM animals WHERE weight > 10.5;
 
-SELECT name FROM animals WHERE date_part('year',date_of_birth) >= 2016 AND date_part('year',date_of_birth) <= 2019;
\ No newline at end of file
+SELECT name FROM animals WHERE date_part('year',date_of_birth) >= 2016 AND date_part('year',date_of_birth) <= 2019;
+
+--Microverse 2nd exercise 
+
+--Delete all data from animals table then rollback 
+BEGIN TRANSACTION;
+TRUNCATE animals;
+ROLLBACK;
+
+--Update animals table with the new species profile
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE RIGHT(name, 3) = 'mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+--Perform delete transactions using rollbacks and savepoints
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT POINT1;
+UPDATE animals SET weight = (weight * -1);
+ROLLBACK TO POINT1;
+UPDATE animals SET weight = (weight * -1) WHERE weight < 0;
+COMMIT;
+
+--Answer to query questions
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight) FROM animals;
+
+SELECT MAX(escape_attempts), neutered FROM animals GROUP BY neutered;
+
+SELECT MAX(weight), MIN(weight), species FROM animals GROUP BY species;
+
+SELECT AVG(escape_attempts), species  FROM animals WHERE date_part('year',date_of_birth) >= 1990 AND da
+te_part('year',date_of_birth) <= 2000 GROUP BY species;
```

Comment:
- [ ] I have checked to confirm that the "setting the species column to unspecified" as provided in the project requirements has not been covered. Kindly include it.

![image](https://user-images.githubusercontent.com/86318284/189984118-b1a192d7-9ceb-4a77-9c7b-fb7657329326.png)

Reply:
@Nemwel-Boniface Thank you for the review, according to this article the newer version does not allow the provision of unknown and unspecified data types, as it will cause errors and problems in recovery and migration. pg_restore will throw an error when running that operation. Details are mentioned here. 

https://dba.stackexchange.com/questions/241870/postgres-column-type-unknown

When trying to alter the columns to the unspecified data type, PostgreSQL throws an error as it is not permissible to make such a conversion at all. 

![error](https://user-images.githubusercontent.com/57537718/190139069-93b4e950-0658-45f4-9ce9-d6f941b5301d.png)